File: 1448-maximum-69-number/maximum-69-number.cpp
class Solution {
public:
    int maximum69Number (int num) {
        int maximum = num;
        string s = to_string(num);
        int length = s.size();

        for (int i = length - 1; i >= 0; --i) {
            int temp = num;
            if (s[length - i - 1] == '6') {
                temp = temp + pow(10, i) * 3;
            }
            if (temp > maximum) {
                maximum = temp;
            }
        }
        return maximum;
    }
};

File: 1448-maximum-69-number/maximum-69-number.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        maximum = num
        length = len(str(num))

        for i in range(length-1,-1,-1):
            temp = num
            if str(num)[length-i-1] == '6':
                temp = temp+(10**i)*3
            
            if temp > maximum:
                maximum = temp
        
        return maximum
            
            